source/unified-test-format/unified-test-format.rst

```diff
@@ -195,6 +195,19 @@ error if an attempt is made to store an entity with a name that already exists
 in the map and MUST raise an error if an entity is not found for a name or is
 found but has an unexpected type.
 
+Test runners MUST provide a mechanism to retrieve entities from the entity
+map prior to the clearing of the entity map, as discussed in
+`Executing a Test`_. There MUST be a way to retrieve an entity by its name
+(for example, to support retrieving the iteration count stored by the
+``storeIterationsAsEntity`` option). Test runners MAY permit or require
+constraining the entity to be retrieved with its expected type.
```

Comment:
Having trouble parsing this sentence after "constraining". Trying to break this down: what does "require constraining the (retrieved) entity with its expected type" mean?

The paragraph that follows talks about allowing test runners to _not_ share certain driver objects in the entity map. For the remaining objects that _are_ shared, is there anything to "constrain"?

Reply:
What I meant by this was when retrieving the entity, the test runner may require to be told both the type and the name of the entity to be retrieved, rather than the name only. It's not strictly needed but I thought it could be a useful additional diagnostic.

Reply:
In PHP's case, I'd expect for my workload executor to fetch the entity and _then_ assert that it's the expected type before I attempt to work with it -- just in the habit of defensive programming. But this text doesn't make me any more or less inclined to do that.

I'll have to defer to stakeholder working with a statically typed languages to chime in on this. My thinking if that they might not have a choice in the matter.

Reply:
My understanding is you want the type verification to be done by the workload executor. The language here permits type verification to be done by the test runner which is how I implemented it.

Reply:
I don't feel strongly about how other languages implement this. My opinion was just that this sentence wasn't necessary to grant Ruby permission to do this. [Entity Map](https://github.com/mongodb/specifications/blob/master/source/unified-test-format/unified-test-format.rst#entity-map) already gives languages a lot of leeway:

> Test runners MAY choose to implement the entity map in a fashion most suited to their language, but implementations MUST enforce both uniqueness of entity names and referential integrity when fetching an entity.

If no one else is going to chime in on this, feel free to resolve this thread after reading and leave the text as-is.

Reply:
I removed it.